libkbfs/journal_server.go

```diff
@@ -311,6 +325,44 @@ func (j *JournalServer) Disable(ctx context.Context, tlfID TlfID) (
 	return true, nil
 }
 
+func (j *JournalServer) deleteTLFJournal(tlfID TlfID) (*tlfJournal, bool) {
+	j.lock.Lock()
+	defer j.lock.Unlock()
+	tlfJournal, ok := j.tlfJournals[tlfID]
+	if !ok {
+		return nil, false
+	}
+	delete(j.tlfJournals, tlfID)
+	return tlfJournal, true
+}
+
+// DisableAndFlush turns off the write journal for the given TLF, and
+// makes sure everything in it is flushed.
+func (j *JournalServer) DisableAndFlush(ctx context.Context, tlfID TlfID) (
+	wasEnabled bool, err error) {
+	j.log.CDebugf(ctx, "Disabling journal for %s", tlfID)
+	defer func() {
+		if err != nil {
+			j.deferLog.CDebugf(ctx,
+				"Error when disabling journal for %s: %v",
+				tlfID, err)
+		}
+	}()
+
+	tlfJournal, ok := j.deleteTLFJournal(tlfID)
```

Comment:
What do you mean "disappear from view"?  The tlfJournal background work is still running, so the pending operations will still complete.  All we're doing is making sure that the next operations hit the real servers, not the journal, until it's re-enabled.  (There won't be any file writes, since mdWriterLock is taken -- only archives/removes.)